sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreReader.java

```diff
@@ -783,6 +784,11 @@ StoreResult createStoreResult(StoreResponse storeResponse,
                 backendLatencyInMs = Double.parseDouble(headerValue);
             }
 
+            headerValue = storeResponse.getHeaderValue(HttpConstants.HttpHeaders.RETRY_AFTER_IN_MILLISECONDS);
+            if (!Strings.isNullOrEmpty(headerValue)) {
+                retryAfterInMs = Double.parseDouble(headerValue);
+            }
```

Comment:
Just a nit since the other null checks seem to be done in a single line, LGTM otherwise!
```suggestion
            if ((headerValue = storeResponse.getHeaderValue(HttpConstants.HttpHeaders.RETRY_AFTER_IN_MILLISECONDS)) != null {
                retryAfterInMs = Double.parseDouble(headerValue);
            };
```